drivers/shared/executor/executor_linux.go

```diff
@@ -677,6 +677,10 @@ func configureIsolation(cfg *runc.Config, command *ExecCommand) error {
 		cfg.Devices = append(cfg.Devices, devs...)
 	}
 
+	for _, device := range cfg.Devices {
+		cfg.Cgroups.Resources.Devices = append(cfg.Cgroups.Resources.Devices, &device.Rule)
+	}
```

Comment:
We've moved this so that it's happening after we've added the user-specified devices (from the [`device`]() block) to the list of default devices from libcontainer (ref [`spec_linux.go#L177-L320`](https://github.com/opencontainers/runc/blob/v1.1.12/libcontainer/specconv/spec_linux.go#L177-L320)). That seems right but not directly motivated by the example... unless you've got a `/dev/fuse` device driver?

Reply:
I'm not entirely sure I've understood your question but I'll take a shot.

To have /dev/fuse usable inside a container, assuming the device is already present in the host, you can add an entry to the 'Devices' field in the executor configuration.

https://github.com/hashicorp/nomad/blob/main/drivers/shared/executor/executor.go#L141

With an entry there (and this PR) you will then have a usable /dev/fuse in the container. It's then possible to use fusermount to mount a filesystem via fuse.

An application where this matters is for [python XARs](https://github.com/facebookincubator/xar). Fuse mounting the XAR makes a huge performance difference when the archive is large and on a networked filesystem.

We are in fact using a custom driver plugin, but it's little more than a thin wrapper around the exec driver. It doesn't do anything special related to devices aside from populating the 'Devices' field.

Let me know if I've missed what you're after.

Reply:
Also added a changelog entry as requested. Thanks for having a look!